### PR TITLE
DEV: Convert group-default-notifications modal to component-based API

### PR DESCRIPTION
**Pull request by janzenisaac**
This PR converts the `group-default-notifications` modal to make use of the new component-based API

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/group-manage-save-button.js [hunk 1]
```diff
@@ -38,53 +41,61 @@ export default Component.extend({
...
+      })
+      .catch((error) => {
+        const json = error.jqXHR.responseJSON;
+        if (error.jqXHR.status === 422 && json.user_count) {
+          this.editGroupNotifications(json);
+        } else {
+          popupAjaxError(error);
+        }
+      })
+      .finally(() => this.set("saving", false));
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/group-manage-save-button.js:88]: `saving` doesn't seem to be used anywhere?

File: app/assets/javascripts/discourse/app/components/group-manage-save-button.js [hunk 2]
```diff
@@ -38,53 +41,61 @@ export default Component.extend({
...
-          }
-        })
-        .finally(() => this.set("saving", false));
-    },
+  @action
+  async editGroupNotifications(json) {
+    await this.modal.show(GroupDefaultNotificationsModal, {
+      model: {
+        count: json.user_count,
+        setUpdateExistingUsers: (e) => this.setUpdateExistingUsers(e),
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/group-manage-save-button.js:96]:
```suggestion
        setUpdateExistingUsers: this.setUpdateExistingUsers,
```

File: app/assets/javascripts/discourse/app/components/group-manage-save-button.js [hunk 3]
```diff
@@ -38,53 +41,61 @@ export default Component.extend({
...
+      },
+    });
+    this.send("save");
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/group-manage-save-button.js:99]:
```suggestion
    this.save();
```

File: app/assets/javascripts/discourse/app/components/group-manage-save-button.js (hunk 1 above)
Comment (janzenisaac) [reply app/assets/javascripts/discourse/app/components/group-manage-save-button.js:88]: We use this for the `savingText`

https://github.com/discourse/discourse/pull/22521/files#diff-2bf398903ccc368349114cd35a8d6d36fced1a77f3d3ff487b5641dea72e0323R18-R21

**Review by CvX — APPROVED** [review]

**Pull request merged**